### PR TITLE
billy/789-dev-error-collection-popup

### LINKED ISSUE #789: new toggle component on preferences page doesn't toggle
console.logs `Analytics disabled in browser` on every click. no visual change or other errors. but seems not to be working.

### PR DESCRIPTION
**Pull request by okwme**
Closes #789 

Description: added popup

screenshot 2018-06-06 13 09 51

### CODE REVIEW

File: app/src/renderer/vuex/modules/user.js [hunk 1]
```diff
@@ -98,12 +98,19 @@ export default ({ commit, node }) => {
...
         localStorage.getItem(`${ERROR_COLLECTION_KEY}_${account}`) === "true"
-      dispatch("setErrorCollection", { account, optin: errorCollection })
+      if (state.errorCollection !== errorCollection)
+        dispatch("setErrorCollection", { account, optin: errorCollection })
     },
-    setErrorCollection({ state }, { account, optin }) {
+    setErrorCollection({ state, commit }, { account, optin }) {
+      if (state.errorCollection !== optin && config.development) {
+        commit("notifyError", {
+          title: `Couldn't switch ${optin ? "on" : "off"} error collection`,
```
Comment (NodeGuy) [inline app/src/renderer/vuex/modules/user.js]: Let's put periods at the end of sentences.

File: app/src/renderer/vuex/modules/user.js [hunk 2]
```diff
@@ -98,12 +98,19 @@ export default ({ commit, node }) => {
...
+          body: "Error collection is disabled during development"
```
Comment (NodeGuy) [inline app/src/renderer/vuex/modules/user.js]: Let's put periods at the end of sentences.

File: app/src/renderer/vuex/modules/user.js [hunk 3]
```diff
@@ -98,12 +98,19 @@ export default ({ commit, node }) => {
...
+        })
+      }
+      optin = config.development ? false : optin
       localStorage.setItem(`${ERROR_COLLECTION_KEY}_${account}`, optin)
-      state.errorCollection = config.development ? false : optin
-
+      state.errorCollection = optin
```
Comment (NodeGuy) [inline app/src/renderer/vuex/modules/user.js]: Let's tweak this to make it more functional:

```
state.errorCollection = config.development ? false : optin
localStorage.setItem(`${ERROR_COLLECTION_KEY}_${account}`, state.errorCollection)
```

File: app/src/renderer/vuex/modules/user.js (hunk 1 above)
Comment (okwme) [reply app/src/renderer/vuex/modules/user.js]: Sounds good to me.

File: app/src/renderer/vuex/modules/user.js (hunk 2 above)
Comment (okwme) [reply app/src/renderer/vuex/modules/user.js]: Sounds good to me.

File: app/src/renderer/vuex/modules/user.js (hunk 3 above)
Comment (okwme) [reply app/src/renderer/vuex/modules/user.js]: that's much nicer

**Pull request merged**